bundles/pubsub/pubsub_admin_tcp/src/pubsub_tcp_handler.c

```diff
@@ -501,10 +496,13 @@ static inline int pubsub_tcpHandler_closeConnectionEntry(
             }
         }
         if (entry->fd >= 0) {
+
+            celixThreadRwlock_unlock(&handle->dbLock);
```

Comment:
Perhaps a correct change, but previously the callbacks would lock, and at the end unlock. Now there is an unlock, and a lock afterwards again. Is this correct?

Reply:
So the possible issue this might introduce is that entry might be free'd in the meantime, when this particular function releases control of the mutex. But the issue with having the lock while doing the `receiverDisconnectMessageCallback`, is that it might give deadlocks due to different lock orders of mutexes. That's why the previous method was to sometimes instruct `receiverDisconnectMessageCallback` to not lock. But this gives reproducable data races due to modifying a hashmap in multiple threads.

Really the solution here would be to use one mutex for both the admin and the receiver/subscribers, at the expense of latency/throughput. But I honestly don't know of any other foolproof way to fix these issues.

Reply:
I see, let's see how this behaves then.